dbms/src/Storages/MergeTree/MergeTreeBaseBlockInputStream.cpp

```diff
@@ -233,6 +233,16 @@ Block MergeTreeBaseBlockInputStream::readFromPart()
               */
             if (constant_filter_description.always_false)
             {
+	            /*
+                  If this filter is PREWHERE 0, MergeTree Stream can be marked as done,
+                  and this task can be clear. I don't understand why below commented out
+                  code is needed. If we don't mark this task finished here, readImpl could
+        		  jump into endless loop.
+                  Error scenario:
+                  select * from table where isNull(NOT_NULLABLE_COLUMN) AND OTHER PRED;
```

Comment:
I made a fix on such scenario. There is a [test](https://github.com/yandex/ClickHouse/blob/master/dbms/tests/queries/0_stateless/00495_reading_const_zero_column.sql).
It will be great if you provide a similar test which which was fixed. Otherwise, it's unclear why we need this change.

Reply:
This symptom depends on data, I am trying to create a simple reproduce, but not right now. But I can consistently encounter this issue on one 805.58 million rows' table.